ecosystem/typescript/sdk_v2/src/crypto/secp256k1.ts

```diff
@@ -0,0 +1,224 @@
+  private readonly key: Hex;
+
+  /**
+   * Create a new PublicKey instance from a Uint8Array or String.
+   *
+   * @param args.hexInput A HexInput (string or Uint8Array)
+   */
+  constructor(args: { hexInput: HexInput }) {
+    super();
+
+    const hex = Hex.fromHexInput(args);
+    if (hex.toUint8Array().length !== Secp256k1PublicKey.LENGTH) {
+      throw new Error(`PublicKey length should be ${Secp256k1PublicKey.LENGTH}`);
+    }
+    this.key = hex;
+  }
+
+  /**
+   * Get the public key in bytes (Uint8Array).
+   *
+   * @returns Uint8Array representation of the public key
+   */
+  toUint8Array(): Uint8Array {
+    return this.key.toUint8Array();
+  }
+
+  /**
+   * Get the public key as a hex string with the 0x prefix.
+   *
+   * @returns string representation of the public key
+   */
+  toString(): string {
+    return this.key.toString();
+  }
+
+  /**
+   * Verifies a signed data with a public key
+   *
+   * @param args.message message
+   * @param args.signature The signature
+   * @returns true if the signature is valid
+   */
+  verifySignature(args: { message: HexInput; signature: Secp256k1Signature }): boolean {
+    const { message, signature } = args;
+    const msgHex = Hex.fromHexInput({ hexInput: message }).toUint8Array();
+    const sha256Message = sha256(msgHex);
+    const rawSignature = signature.toUint8Array();
+    return secp256k1.verify(rawSignature, sha256Message, this.toUint8Array());
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeBytes(this.key.toUint8Array());
+  }
+
+  static deserialize(deserializer: Deserializer): PublicKey {
+    const bytes = deserializer.deserializeBytes();
+    return new Secp256k1PublicKey({ hexInput: bytes });
+  }
+}
+
+/**
+ * A Secp256k1 ecdsa private key
+ */
+export class Secp256k1PrivateKey extends PrivateKey {
+  /**
+   * Length of Secp256k1 ecdsa private key
+   */
+  static readonly LENGTH: number = 32;
+
+  /**
+   * The private key bytes
+   * @private
+   */
+  private readonly key: Hex;
+
+  /**
+   * Create a new PrivateKey instance from a Uint8Array or String.
+   *
+   * @param args.hexInput A HexInput (string or Uint8Array)
+   */
+  constructor(args: { hexInput: HexInput }) {
+    super();
+
+    const privateKeyHex = Hex.fromHexInput(args);
+    if (privateKeyHex.toUint8Array().length !== Secp256k1PrivateKey.LENGTH) {
+      throw new Error(`PrivateKey length should be ${Secp256k1PrivateKey.LENGTH}`);
+    }
+
+    this.key = privateKeyHex;
+  }
+
+  /**
+   * Get the private key in bytes (Uint8Array).
+   *
+   * @returns
+   */
+  toUint8Array(): Uint8Array {
+    return this.key.toUint8Array();
+  }
+
+  /**
+   * Get the private key as a hex string with the 0x prefix.
+   *
+   * @returns string representation of the private key
+   */
+  toString(): string {
+    return this.key.toString();
+  }
+
+  /**
+   * Sign the given message with the private key.
+   * The `lowS` option is default to `true`
```

Comment:
should we provide an option to set the `lowS`?
```suggestion
   * The `lowS` option is default to `true`
```

Reply:
Base on what I am seeing, we always set this to true. I think we should be fine here.

Reply:
You should remove `lowS` from here. Signing should always produce signatures with low `s` since that's what our blockchain exclusively accepts.

Reply:
Sounds good. Will remove from comment